### PR TITLE
[BEAM-10112] Add more state and timer python examples to website

### PR DESCRIPTION
**Pull request by y1chi**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Dataflow/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by y1chi** [comment]
R: @angoenka

**Comment by y1chi** [comment]
R: @boyuanzz

**Comment by boyuanzz** [comment]
r: @rosetn

**Comment by y1chi** [comment]
@rosetn friendly ping for review

**Review by boyuanzz — APPROVED** [review]
Looks good from my side.

**Comment by rosetn** [comment]
Run Website_Stage_GCS PreCommit

File: website/www/site/content/en/documentation/programming-guide.md [hunk 1]
```diff
@@ -4519,6 +4519,8 @@ registered, then Beam will automatically infer the coder for the state value. Ot
 specified when creating the ValueState. For example, the following ParDo creates a  single state variable that
 accumulates the number of elements seen.
 
+Note: ValueState is called ReadModifyWriteState in Python SDK.
```
Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4522]: Code font `ValueState` and `ReadModifyWriteState`

File: website/www/site/content/en/documentation/programming-guide.md [hunk 2]
```diff
@@ -4544,6 +4546,19 @@ perUser.apply(ParDo.of(new DoFn<KV<String, ValueT>, OutputT>() {
 }));
 {{< /highlight >}}
 
+{{< highlight python >}}
```
Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4549]: This Python section does not show up when I have Python selected.

http://apache-beam-website-pull-requests.storage.googleapis.com/13591/documentation/programming-guide/index.html#state-and-timers

File: website/www/site/content/en/documentation/programming-guide.md [hunk 3]
```diff
@@ -4775,12 +4795,15 @@ _ = (p | 'Read per user' >> ReadPerUser()
 
 #### 11.3.3. Dynamic timer tags {#dynamic-timer-tags}
 
-Beam also supports dynamically setting a timer tag using `TimerMap`. This allows for setting multiple different timers
+Beam also supports dynamically setting a timer tag using `TimerMap` in Java SDK. This allows for setting multiple different timers
```
Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4798]: "in the Java SDK."

File: website/www/site/content/en/documentation/programming-guide.md [hunk 4]
```diff
@@ -4775,12 +4795,15 @@ _ = (p | 'Read per user' >> ReadPerUser()
...
 in a `DoFn` and allowing for the timer tags to be dynamically chosen - e.g. based on data in the input elements. A
 timer with a specific tag can only be set to a single timestamp, so setting the timer again has the effect of
 overwriting the previous expiration time for the timer with that tag. Each `TimerMap` is identified with a timer family
 id, and timers in different timer families are independent.
 
+In Python SDK, a dynamic timer tag can be specified while calling timer.set() or timer.clear(), by default the timer
```
Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4804]: "In the Python SDK"

Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4804]: `set()` and `clear()`

Unless you think it's too ambiguous without the class name. If that's the case, then add these in code font.

Comment (rosetn) [inline website/www/site/content/en/documentation/programming-guide.md:4804]: I think these are two sentences? "... or `clear()`. By default, the timer..."

**Review by rosetn — COMMENTED** [review]
Hey Yichi--your Python highlights sections are not rendering. You can view the staged site here: http://apache-beam-website-pull-requests.storage.googleapis.com/13591/documentation/programming-guide/index.html#state-and-timers

File: website/www/site/content/en/documentation/programming-guide.md (hunk 1 above)
Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4522]: Done.

File: website/www/site/content/en/documentation/programming-guide.md (hunk 3 above)
Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4798]: Done.

File: website/www/site/content/en/documentation/programming-guide.md (hunk 2 above)
Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4549]: Done.

File: website/www/site/content/en/documentation/programming-guide.md (hunk 4 above)
Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4804]: Done.

Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4804]: Done.

Comment (y1chi) [reply website/www/site/content/en/documentation/programming-guide.md:4804]: Done.

**Comment by y1chi** [comment]
@rosetn thanks for reviewing, I've addressed the comments, PTAL.

**Review by rosetn — APPROVED** [review]

**Comment by boyuanzz** [comment]
Thanks! I'm going to merge this PR.

**Pull request merged**